portability-api-launcher/src/main/java/org/dataportabilityproject/api/launcher/ExtensionContext.java

```diff
@@ -43,11 +43,13 @@
    */
   default <T> void registerService(Class<T> type, T service) {}
 
+  CommonSettings getCommonSettings();
+
   /**
-   * Returns the configuration parameter for the key or the default value if not found.
+   * Returns the configuration value for {@code setting} or the default value if not found.
    *
-   * @param key the parameter key
+   * @param setting the parameter key
    * @param defaultValue the default value. Null may be passed.
    */
-  <T> T getConfiguration(String key, T defaultValue);
+  <T> T getSetting(String setting, T defaultValue);
```

Comment:
It this just  pass through to settingsExtension each time? If so, do we need to specify a method or constructor that requires (SettingsExtension) method of some type?

Reply:
When will a developer use this vs getConfiguration?